### PR TITLE
Auth: Make GitHub auth's allowed_organizations be case insensitive

### LINKED ISSUE #66876: Auth: GitHub auth's `allowed_organizations` is case sensitive, but shouldn't be
## What happened:

Configuring GitHub authentication with the [allowed_organizations](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/github/#allowed_organizations) option works but is case sensitive when it shouldn't be.

### Error seen by user

image

### Grafana logs

```
logger=context userId=0 orgId=0 uname= t=2023-04-19T15:36:57.331244874Z level=info msg="Request Completed" method=GET path=/login/github status=302 remote_addr=192.168.51.250 time_ms=10 duration=10.683155ms size=287 referer=https://grafana.mydomain.org/login handler=/login/:name
logger=oauth t=2023-04-19T15:36:57.872391808Z level=info msg="state check" queryState=1f28105009e8e0fb1382e2519bab9e52f72c13f151ca1b3b6f2425c693c2ef09 cookieState=1f28105009e8e0fb1382e2519bab9e52f72c13f151ca1b3b6f2425c693c2ef09
logger=oauth.github t=2023-04-19T15:36:58.441988073Z level=warn msg="No valid role found. Skipping role sync. In Grafana 10, this will result in the user being assigned the default role and overriding manual assignment. If role sync is not desired, set oauth_skip_org_role_update_sync to true"
logger=context userId=0 orgId=0 uname= t=2023-04-19T15:36:58.515119383Z level=error msg="login.OAuthLogin(get info from github)" error="user not a member of one of the required organizations"
logger=context userId=0 orgId=0 uname= t=2023-04-19T15:36:58.515299956Z level=error msg="Request Completed" method=GET path=/login/github status=500 remote_addr=192.168.51.250 time_ms=653 duration=653.439725ms size=1365 referer=https://mydomain.org.cloud/ handler=/login/:name
```

## What you expected to happen:

Configuring `allowed_organizations` with `MyGitHubOrganization` or `mygithuborganization` shouldn't matter as github doesn't distinguish these, but the grafana github authenticator does.

For reference, [these API docs](https://docs.github.com/en/github-ae@latest/rest/orgs/members#check-organization-membership-for-a-user) mentions that the case of the github org doesn't matter.

image

## How to reproduce it (as minimally and precisely as possible):

I think GitHub auth needs to be configured fully, with a github OAuth application's client_id and client_secret, and then use of `allowed_organizations` configured with that.

## Environment:

- Grafana version: 9.4.3
  Looking through the changelog, I [see no bugfix about this between the latest release 9.4.7 and 9.4.3](https://github.com/grafana/grafana/blob/main/CHANGELOG.md)
- Data source type & version: Not relevant
- OS Grafana is installed on: Linux, via the official grafana helm chart versioned 6.52.2
- User OS & Browser: Linux, Chrome
- Grafana plugins: None
- Others:

### PR DESCRIPTION
**Pull request by consideRatio**
**What is this feature?**

Its a bugfix for the github authentication for the `allowed_organizations` configuration that incorrectly was case sensitive. Some github organizations have a capitcal case, for example https://github.com/smithsonian, which can be seen using the GitHub CLI to work against GitHub's API:

```
gh api /orgs/smithsonian | jq .login
"Smithsonian"
```

**Which issue(s) does this PR fix?**:

Fixes #66876

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by mgyongyosi — APPROVED** [review]
Hey @consideRatio, thank you for this PR!

File: pkg/login/social/github_oauth.go [hunk 1]
```diff
@@ -67,7 +68,7 @@ func (s *SocialGithub) IsOrganizationMember(client *http.Client, organizationsUr
 
 	for _, allowedOrganization := range s.allowedOrganizations {
 		for _, organization := range organizations {
-			if organization == allowedOrganization {
+			if strings.ToLower(organization) == strings.ToLower(allowedOrganization) {
```
Comment (mgyongyosi) [inline pkg/login/social/github_oauth.go:71]: I have just seen the linter error. Could you please change this part to use the `strings.EqualFold()` function?

**Review by mgyongyosi — CHANGES_REQUESTED** [review]

File: pkg/login/social/github_oauth.go (hunk 1 above)
Comment (consideRatio) [reply pkg/login/social/github_oauth.go]: Ah, interesting! Done in 1d221fe!

**Review by mgyongyosi — APPROVED** [review]

**Comment by mgyongyosi** [comment]
@consideRatio Thanks for the PR! It has been merged! 🚀

**Pull request merged**